include/parser/riscv/riscv-regex.hpp

```diff
@@ -0,0 +1,125 @@
+/* C++ Assembler Interpreter
+ * Copyright (C) 2016 Chair of Computer Architecture
+ * at Technical University of Munich
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.*/
+
+#ifndef ERAGPSIM_PARSER_RISCV_RISCV_REGEX_HPP
+#define ERAGPSIM_PARSER_RISCV_RISCV_REGEX_HPP
+
+#include <regex>
+#include <string>
+
+#include "parser/riscv/riscv-parser.hpp"
+
+class CompileErrorList;
+#include "parser/independent/positioned-string.hpp"
+
+/**
+ * Wrapper class for the RegExp used for parsing.
+ */
+class RiscvParser::RiscvRegex {
+ public:
+  RiscvRegex();
+
+  /**
+   * Returns if the line is a valid Risc-V instruction.
+   */
+  bool isValid() const noexcept;
+
+  /**
+   * Returns if the line contains a label.
+   */
+  bool hasLabel() const noexcept;
+
+  /**
+   * Returns if the line contains an instruction.
+   */
+  bool hasInstruction() const noexcept;
+
+  /**
+   * Returns if the instruction is a parser directive.
+   */
+  bool isDirective() const;
+
+  /**
+   * Returns the parsed label.
+   */
+  const PositionedString &getLabel() const noexcept;
+
+  /**
+   * Returns the parsed instruction.
+   */
+  PositionedString getInstruction() const;
+
+  /**
+   * Returns the n-th parameter of the parsed instruction.
+   *
+   * \param n Zero-based index of the parameter
+   */
+  const PositionedString &getParameter(int n) const;
+
+  /**
+   * Returns the number of parameters found.
+   */
+  int getParameterCount() const noexcept;
+
+  /**
+   * Parses the next line.
+   *
+   * \param line  The line to parse.
+   * \param lineCoordinate
+   * \param errors CompileErrorList to record errors.
+   * \return The last read position in the string (if the line does not match,
+   * this is the character that caused the error)
+   */
+  size_t matchLine(const std::string &line,
+                   CodeCoordinate lineCoordinate,
+                   CompileErrorList &errors);
+
+ private:
+  bool _readInstructionOrLabel(const std::string &line,
```

Comment:
Private methods should also have documentation, for future developers and yourself